Source/WebCore/platform/graphics/mac/controls/ControlFactoryMac.h

```diff
@@ -35,6 +35,7 @@ OBJC_CLASS NSServicesRolloverButtonCell;
 namespace WebCore {
 
 class ControlFactoryMac final : public ControlFactoryCocoa {
+    friend class SearchFieldResultsMac;
```

Comment:
Why is this necessary?

Reply:
To make `SearchFieldResultsMac` have access to the new private method `ControlFactoryMac::searchMenuTemplate()`. I did not want to make it public since it is only used by `SearchFieldResultsMac::updateCellStates()`.

Reply:
I see. Can `searchMenuTemplate()` be a private method on `SearchFieldResultsMac`?

I generally prefer to avoid use of `friend` when possible.

Reply:
Or alternately, pass it in to the constructor?

Reply:
`searchMenuTemplate` is now passed to `SearchFieldResultsMac` constructor as an argument.